### PR TITLE
Add note about the store retaining the query watcher

### PR DESCRIPTION
**Pull request by designatednerd**
Trying to add a bit more information about what's happening until I can get around to fixing #516.

### CODE REVIEW

**Comment by designatednerd** [comment]
@AnthonyMDev @koenpunt @wnagrodzki would appreciate feedback on the language here

File: Sources/Apollo/GraphQLQueryWatcher.swift [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import Dispatch
 
 /// A `GraphQLQueryWatcher` is responsible for watching the store, and calling the result handler with a new result whenever any of the data the previous result depends on changes.
+///
+/// NOTE: The store retains the watcher and does not automatically unsubscribe you. You must call `cancel()` on your query watcher when you no longer need results.
```
Comment (AnthonyMDev) [inline Sources/Apollo/GraphQLQueryWatcher.swift]:
```suggestion
/// NOTE: The store retains the watcher while subscribed. You must call `cancel()` on your query watcher when you no longer need results. Failure to call `cancel()` before releasing your reference to the returned watcher will result in a memory leak.
```

**Comment by wnagrodzki** [comment]
Thank you. I think that's perfect :)

What do you think about updating [this tutorial page](https://www.apollographql.com/docs/ios/caching/#watching-queries) with as well? Maybe replacing _kill of_ with mention about `cancel()` method and memory leak will draw a bit more attention to this issue for newcomers.

> Remember to kill off watchers when their parent object is deallocated! This is not (presently) done automatically.

**Comment by designatednerd** [comment]
@wnagrodzki Yep great point.

**Pull request merged**